Panic when a packs `variables.hcl` file is missing

While developing a new pack, I ran into the panic below when my packs `variables.hcl` file was misnamed `variable.hcl`.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xea0541]

goroutine 1 [running]:
github.com/hashicorp/nomad-pack/internal/pkg/variable.(*Parser).loadPackFile(0xc0001171d0, 0x0, 0xc00031f790, 0xc, 0x0, 0x0, 0x16811d0)
        /home/circleci/project/project/internal/pkg/variable/parser.go:152 +0xa1
github.com/hashicorp/nomad-pack/internal/pkg/variable.(*Parser).parseRootFiles(0xc0001171d0, 0x7f20baa67108, 0x8, 0xc0007b60b0)
        /home/circleci/project/project/internal/pkg/variable/root.go:14 +0x125
github.com/hashicorp/nomad-pack/internal/pkg/variable.(*Parser).Parse(0xc0001171d0, 0xc0001171d0, 0x0, 0x0, 0xa)
        /home/circleci/project/project/internal/pkg/variable/parser.go:81 +0x45
github.com/hashicorp/nomad-pack/internal/pkg/manager.(*PackManager).ProcessTemplates(0xc00000c5b8, 0x18, 0x12f5920, 0x7ffd25b11d01, 0xc00000c5b8)
        /home/circleci/project/project/internal/pkg/manager/manager.go:80 +0x3fd
github.com/hashicorp/nomad-pack/cli.renderPack(0xc00000c5b8, 0x167e6e8, 0xc00079c300, 0xc00031fca8, 0x0, 0x0, 0x2)
        /home/circleci/project/project/cli/helpers.go:235 +0x45
github.com/hashicorp/nomad-pack/cli.(*RenderCommand).Run(0xc00000c390, 0xc00018c020, 0x1, 0x1, 0xc00020f2e0)
        /home/circleci/project/project/cli/render.go:55 +0x4c5
github.com/mitchellh/cli.(*CLI).Run(0xc0007a9540, 0xc00031fef8, 0xc00000e2e8, 0x1)
        /go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x41a
github.com/hashicorp/nomad-pack/cli.Main(0xc00018c000, 0x3, 0x3, 0x0)
        /home/circleci/project/project/cli/main.go:94 +0x225
main.main()
        /home/circleci/project/project/main.go:12 +0xa5
```

I just checked and the changes made in #113 to fix #112 appear to resolve this issue.

```
 ➜  csi-cinder git:(main) ✗ /workspaces/nomad-pack/bin/nomad-pack render csi-cinder
! Failed To Process Pack

        Error:   This variable value is not compatible with the variable's type constraint: attribute "enabled" is required.
        Type:    *errors.errorString
        Context: 
                 - HCL Range: /root/.nomad/packs/default/csi-cinder@latest/variables.hcl:42,13-48,4
                 - Registry Name: default
                 - Pack Name: csi-cinder
                 - Pack Ref: latest
```

--- a/internal/pkg/manager/manager.go
+++ b/internal/pkg/manager/manager.go
@@ -114,8 +114,8 @@ func (pm *PackManager) loadAndValidatePacks() (*pack.Pack, error) {
 		return nil, fmt.Errorf("failed to load pack: %v", err)
 	}
 
-	if err := parentPack.Metadata.Validate(); err != nil {
-		return nil, fmt.Errorf("failed to validate pack metadata: %v", err)
+	if err := parentPack.Validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate pack: %v", err)
 	}
 
 	// Using the input path to the parent pack, define the path where
@@ -146,8 +146,8 @@ func (pm *PackManager) loadAndValidatePack(cur *pack.Pack, depsPath string) erro
 			return fmt.Errorf("failed to load dependent pack: %v", err)
 		}
 
-		if err := dependentPack.Metadata.Validate(); err != nil {
-			return fmt.Errorf("failed to validate dependent pack metadata: %v", err)
+		if err := dependentPack.Validate(); err != nil {
+			return fmt.Errorf("failed to validate dependent pack: %v", err)
 		}
 
 		// Add the dependency to the current pack.

--- a/pkg/pack/pack.go
+++ b/pkg/pack/pack.go
@@ -1,5 +1,7 @@
 package pack
 
+import "errors"
+
 // File is an individual file component of a Pack.
 type File struct {
 
@@ -82,4 +84,20 @@ func (p *Pack) RootVariableFiles() map[string]*File {
 	}
 
 	return out
+}
+
+// Validate the pack for terminal problems that can easily be detected at this
+// stage. Anything that has potential to cause a panic should ideally be caught
+// here.
+func (p *Pack) Validate() error {
+
+	if p.RootVariableFile == nil {
+		return errors.New("root variable file is required")
+	}
+
+	if err := p.Metadata.Validate(); err != nil {
+		return err
+	}
+
+	return nil
 }

